src/api/team-membership/content-types/team-membership/schema.json

```diff
@@ -0,0 +1,32 @@
+{
+  "kind": "collectionType",
+  "collectionName": "team_memberships",
+  "info": {
+    "singularName": "team-membership",
+    "pluralName": "team-memberships",
+    "displayName": "TeamMembership",
+    "description": ""
+  },
+  "options": {
+    "draftAndPublish": false
+  },
+  "pluginOptions": {},
+  "attributes": {
+    "user": {
+      "type": "relation",
+      "relation": "oneToOne",
+      "target": "plugin::users-permissions.user"
+    },
+    "joinDate": {
+      "type": "datetime"
+    },
+    "leaveDate": {
+      "type": "datetime"
+    },
+    "project": {
```

Comment:
I think this should be a required field.

Reply:
Okay, I also think I will have to make the default value "null" at first. This is cause the team-membership record for an active user still in a team won't have a `leaveDate` just yet

Reply:
Which, you mean the `leaveDate` or `project` field ?

Reply:
The project field.

Reply:
Done now